pandasai/llm/llama2_text_gen.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional, Any, List, Dict
+
+from .base import LLM
+from ..helpers import load_dotenv
+from ..prompts.base import Prompt
+
+load_dotenv()
+
+
+class LLaMa2TextGen(LLM):
+    """LLaMa-2 Text Generation Inference LLM"""
+
+    max_new_tokens: int = 1024
+    top_k: Optional[int] = None
+    top_p: Optional[float] = 0.8
+    typical_p: Optional[float] = 0.8
+    temperature: float = 1E-3  # must be strictly positive
+    repetition_penalty: Optional[float] = None
+    truncate: Optional[int] = None
+    stop_sequences: List[str] = None
+    seed: Optional[int] = None
+    do_sample: Optional[bool] = False
+    inference_server_url: str = ""
+    streaming: Optional[bool] = False
+    timeout: int = 120
+    use_chat_model: Optional[bool] = False
+    client: Any
+
+    def __init__(self, inference_server_url: str, **kwargs):
+        try:
+            import text_generation
+
+            for (key, val) in kwargs.items():
+                if key in self.__annotations__:
+                    setattr(self, key, val)
+
+            self.client = text_generation.Client(
+                base_url=inference_server_url,
+                timeout=self.timeout,
+            )
+
+        except ImportError:
+            raise ImportError(
+                "Could not import text_generation python package. "
+                "Please install it with `pip install text_generation`."
+            )
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling text generation inference API."""
+        return {
+            "max_new_tokens": self.max_new_tokens,
+            "top_k": self.top_k,
+            "top_p": self.top_p,
+            "typical_p": self.typical_p,
+            "temperature": self.temperature,
+            "repetition_penalty": self.repetition_penalty,
+            "truncate": self.truncate,
+            "stop_sequences": self.stop_sequences,
+            "do_sample": self.do_sample,
+            "seed": self.seed,
+        }
+
+    def call(self, instruction: Prompt, suffix: str = "") -> str:
+        prompt = instruction.to_string() + suffix
+        if self.use_chat_model:
+            # format prompt for LLaMa-2 Chat
+            prompt = f"[INST] <<SYS>>\n{prompt}\n<</SYS>>\n\n"
+
```

Comment:
@zachschillaci27 yep, makes a lot of sense to me. Also, given the fact that this is the only part "llama2-specific" we could make this class a `HuggingfaceTextGeneration` LLM class to support any model and add some tutorials in the docs to explain how to specifically use LLaMa 2, CodeLlaMa, Starcoder etc.

Reply:
@zachschillaci27 that's a great idea, same for WizardCoder etc.. I'm playing around a little bit with it, will keep you posted, but it's definitely a thing that we'll work on ASAP!